File: rest-service/api/src/main/java/org/veupathdb/service/multiblast/Resources.java
package org.veupathdb.service.multiblast;

import org.glassfish.jersey.media.multipart.MultiPartFeature;
import org.veupathdb.lib.container.jaxrs.config.Options;
import org.veupathdb.lib.container.jaxrs.server.ContainerResources;
import org.veupathdb.service.multiblast.controller.DebugController;
import org.veupathdb.service.multiblast.controller.GuestLinkController;
import org.veupathdb.service.multiblast.controller.JobController;
import org.veupathdb.service.multiblast.controller.MetaController;

/**
 * Service Resource Registration.
 *
 * This is where all the individual service specific resources and middleware
 * should be registered.
 */
public class Resources extends ContainerResources {
  public Resources(Options opts) {
    super(opts);
  }

  /**
   * Returns an array of JaxRS endpoints, providers, and contexts.
   *
   * Entries in the array can be either classes or instances.
   */
  @Override
  protected Object[] resources() {
    return new Object[] {
      MultiPartFeature.class,

      JobController.class,
      MetaController.class,
      DebugController.class,
      GuestLinkController.class,
    };
  }
}

File: service-report/src/main/kotlin/org/veupathdb/service/mblast/report/Resources.kt
package org.veupathdb.service.mblast.report

import org.veupathdb.lib.container.jaxrs.config.Options
import org.veupathdb.lib.container.jaxrs.server.ContainerResources
import org.veupathdb.service.mblast.report.controller.*

class Resources(opts: Options) : ContainerResources(opts) {
  init {
    enableAuth()
  }

  override fun resources() =
    arrayOf(
      JobByIDController::class.java,
      JobsController::class.java,
      JobsStdErrController::class.java,
      JobFileListController::class.java,
      JobFileDownloadController::class.java,
      GuestLinkController::class.java,
      CacheMaintenanceController::class.java,
      BrokenJobController::class.java,
    )
}